src/Providers/ElasticApmServiceProvider.php

```diff
@@ -243,6 +248,44 @@ protected function listenForQueries()
         });
     }
 
+    protected function listenForTransactions()
+    {
+        $this->app->events->listen(TransactionBeginning::class, function (TransactionBeginning $transactionBeginning) {
+            self::$dbTransactionStartsByDB[$transactionBeginning->connection->getDatabaseName()][] = microtime(true);
```

Comment:
```suggestion
            self::$dbTransactionStartsByDB[$transactionBeginning->connectionName][] = microtime(true);
```

`ConnectionEvent` abstract class constructor sets this field.